rust/src/tests/lib/aggregator.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+    aggregator::service::{Aggregator, ServiceHandle as InnerServiceHandle, ServiceRequests},
+    common::client::Credentials,
+};
+use bytes::Bytes;
+use futures::future;
+
+#[derive(Clone)]
+pub struct ServiceHandle(InnerServiceHandle);
+
+pub struct ByteAggregator {
+    weights: Vec<u8>,
+}
+
+impl ByteAggregator {
+    pub fn new() -> ByteAggregator {
+        ByteAggregator { weights: vec![] }
+    }
+}
+
+impl Aggregator for ByteAggregator {
+    type Error = ();
+
+    type AddWeightsFut = future::Ready<Result<(), ()>>;
+    type AggregateFut = future::Ready<Result<Bytes, ()>>;
+
+    fn add_weights(&mut self, weights: Bytes) -> Self::AddWeightsFut {
+        weights.into_iter().for_each(|el| self.weights.push(el));
```

Comment:
Sorry for not seeing that earlier but I think this could be simplified:

```
self.weights.extend(weights.into_iter())
```

or something like that.

Reply:
Fixed 😊